docker/package/package_generator.py

```diff
@@ -0,0 +1,141 @@
+
+def gen_makefile(pkg: Package, out):
+    makefile_contents =f'''
+.PHONY: install
+
+BINDIR=/usr/bin
+
+{pkg.name}:
+	./compile.sh
+	cp $(CURDIR)/opam/default/bin/{pkg.name} {pkg.name}
+
+install: {pkg.name}
+	mkdir -p $(DESTDIR)$(BINDIR)
+	cp $(CURDIR)/{pkg.name} $(DESTDIR)$(BINDIR)
+'''
+    with open(out, 'w') as f:
+        f.write(makefile_contents)
+
+def gen_changelog(pkg: Package, ubuntu_version, maintainer, date, out):
+    changelog_contents = f'''{pkg.name.lower()} ({package_version}) {ubuntu_version}; urgency=medium
+
+  * Publish {version}-{release} version of {pkg.name}
+
+ -- {maintainer} {date}'''
+    with open(out, 'w') as f:
+        f.write(changelog_contents)
+
+def gen_rules(out):
+    rules_contents = '''#!/usr/bin/make -f
+
+%:
+	dh $@ '''
+    with open(out, 'w') as f:
+        f.write(rules_contents)
+
+packages = [
+    Package("tezos-client",
+            "CLI client for interacting with tezos blockchain"),
+    # TODO: uncomment, commented to speed up CI builds
+    # Package("tezos-admin-client",
+    #         "Administration tool for the node") ,
+    # Package("tezos-node",
+    #         "Entry point for initializing, configuring and running a Tezos node"),
+    # Package("tezos-signer",
+    #         "A client to remotely sign operations or blocks")
+]
+
+# TODO: uncomment, commented to speed up CI builds
+daemons = ["baker", # "accuser", "endorser"
+]
+
+daemon_decs = {
+    "baker": "Daemon for baking",
+    "accuser": "Daemon for accusing",
+    "endorser": "Daemon for endorsing"
+}
+
+for protocol in active_protocols:
+    for daemon in daemons:
+        packages.append(
+            Package(f"tezos-{daemon}-{protocol}",
+                daemon_decs[daemon]
+            )
+        )
+
+for package in packages:
+    dir = f"{package.name.lower()}-0ubuntu{version}"
+    # tezos-client and tezos-admin-client are in one opam package
+    opam_package = "tezos-client" if package.name == "tezos-admin-client" else package.name
+    subprocess.run(["opam", "exec", "--", "opam-bundle", f"{opam_package}={version}",
+                    "--ocaml=4.09.1", "--yes", "--opam=2.0.5"], check=True)
+    subprocess.run(["tar", "-zxf", f"{opam_package}-bundle.tar.gz"], check=True)
+    os.rename(f"{opam_package}-bundle", dir)
+    gen_makefile(package, f"{dir}/Makefile")
+    subprocess.run(["tar", "-czf", f"{dir}.tar.gz", dir], check=True)
+    os.chdir(dir)
+    subprocess.run(["dh_make", "-syf" f"../{dir}.tar.gz"], check=True)
+    gen_control_file(package, "debian/control")
+    subprocess.run(["wget", "-q", "-O", "debian/copyright", f"https://gitlab.com/tezos/tezos/-/raw/v{version}/LICENSE"], check=True)
+    subprocess.run("rm debian/*.ex debian/*.EX debian/README*", shell=True, check=True)
+    gen_changelog(package, "bionic", meta["maintainer"], date, "debian/changelog")
+    gen_rules("debian/rules")
+    subprocess.run(["dpkg-buildpackage", "-S" if sys.argv[1] == "source" else "-b", "-us", "-uc"],
```

Comment:
Can you define something like `is_source` somewhere above to avoid using `sys.argv[1]` more than once? And I would consider three cases: `source`, `binary` and anything else (we should fail in this case).